File: swift/connect_april.swift
import Foundation

class ConnectApril {
    static func recvRequest() -> [String: Any]? {
        let arguments = CommandLine.arguments
        guard let n = arguments.last else { return nil }
        let filePath = "./\(n).input"

        do {
            let data = try String(contentsOfFile: filePath, encoding: .utf8)
            return try JSONSerialization.jsonObject(with: data.data(using: .utf8)!, options: []) as? [String: Any]
        } catch {
            print(error)
            return nil
        }
    }

    static func sendResponse(response: [String: Any]) {
        let arguments = CommandLine.arguments
        guard let n = arguments.last else { return }
        let filePath = "./\(n).output"

        do {
            let jsonData = try JSONSerialization.data(withJSONObject: response)
            try jsonData.write(to: URL(fileURLWithPath: filePath), options: [])
        } catch {
            print(error)
        }
    }
}

File: kotlin/connect_april.kt
import java.io.File
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import com.fasterxml.jackson.module.kotlin.readValue

class ConnectApril {
    companion object {
        fun recvRequest(): Map<String, Any?>? {
            val n = args.last().toIntOrNull()
            val filePath = if (n != null) ".${n}.input" else return null

            try {
                val data = File(filePath).readText()
                return jacksonObjectMapper().readValue<Map<String, Any?>>(data)
            } catch (e: Exception) {
                println("Error: $e")
                return null
            }
        }

        fun sendResponse(response: Any?) {
            val n = args.last().toIntOrNull()
            val filePath = if (n != null) ".${n}.output" else return

            try {
                File(filePath).writeText(jacksonObjectMapper().writeValueAsString(response))
            } catch (e: Exception) {
                println("Error: $e")
            }
        }
    }
}